pontoon/static/js/widgets/columns.js

```diff
@@ -0,0 +1,97 @@
+
+import React from 'react';
+
+
+export class Columns extends React.Component {
+
+    render() {
+        return (
+            <Container
+               columns={this.columns.length}
+               ratios={this.columns.map(([, v]) => v)}>
+              {this.columns.map(([column,], key) => {
+                  return (
+                      <Column key={key}>
+                        {column}
+                      </Column>);
+              })}
+            </Container>);
+    }
+}
+
+
+export class Container extends React.Component {
+
+    createColumnStyle(width) {
+        return {
+            float: "left",
+            overflow: "hidden",
+            boxSizing: "border-box",
+            width: width.toString() + "%"};
+    }
```

Comment:
I am not a big fan of having CSS properties in JS. Since we're using webpack, it would be quite simple to have a CSS file live next to this, and import it here. Also, I'm wondering if the width problem you're solving with JS could be done with CSS instead?

Reply:
this is a definite exception. This does one thing and one thing only - it gives columns percentage widths in a grid layout - its the whole point of this component